stablehlo/tests/TestUtils.cpp

```diff
@@ -98,8 +109,9 @@ struct InferReturnTypeComponentsPattern : public RewritePattern {
     auto *newOp = rewriter.create(state);
     for (const auto &it : llvm::enumerate(components)) {
       if (it.value().hasRank()) {
-        newOp->setAttr((StringRef("dims") + Twine(it.index())).str(),
-                       rewriter.getI64ArrayAttr(it.value().getDims()));
+        newOp->setAttr(
+            (StringRef("dims") + Twine(it.index())).str(),
+            rewriter.getI64ArrayAttr(fixDynamicDims(it.value().getDims())));
```

Comment:
I think we should switch from -1 to ? here as well. How about `rewriter.getStringAttr(dimSizesToString(it.value().getDims()))`?

Reply:
We could, but that would print as `dim0 = "[0, ?]"` which is also a bit inconsistent with quotes.

The other alternative is registering this op to control printing, or creating a new attr type for dim size that has a custom printer/parser to use for this type. I don't think we should register test ops unless we make a whole new test dialect, and we currently use ArrayRefParam for dim sizes which also feels correct. So I could go either way for this one between -1 and the string above.

Reply:
"I don't think we should register test ops unless we make a whole new test dialect". Agreed.

Between keeping -1 and accepting quotes, I think quotes are preferable, because the MLIR community has really been trying to get away from -1, and the mission has basically succeeded.